peerconnection.go

```diff
@@ -932,49 +1005,64 @@ func (pc *PeerConnection) startRTPReceivers(incomingTracks map[uint32]trackDetai
 		remoteIsPlanB = descriptionIsPlanB(pc.RemoteDescription())
 	}
 
-	// Ensure we haven't already started a transceiver for this ssrc
-	for ssrc := range incomingTracks {
-		for i := range localTransceivers {
-			if t := localTransceivers[i]; (t.Receiver()) == nil || t.Receiver().Track() == nil || t.Receiver().Track().ssrc != ssrc {
+	// Ensure we haven't already started a transceiver for this trackid
+	for incomingTrackID := range incomingTracks {
+		for _, t := range localTransceivers {
+			if t.Receiver() == nil || t.Receiver().Track() == nil {
+				continue
+			}
+
+			// with unified plan track id is the transceiver mid
+			trackID := t.Mid()
+			if remoteIsPlanB {
+				// with planB trackId is the receiver track id
+				trackID = t.Receiver().Track().id
+			}
+			if trackID != incomingTrackID {
 				continue
 			}
 
-			delete(incomingTracks, ssrc)
+			delete(incomingTracks, incomingTrackID)
 		}
 	}
 
-	for ssrc, incoming := range incomingTracks {
+	for incomingTrackID, incoming := range incomingTracks {
 		for i := range localTransceivers {
 			t := localTransceivers[i]
 
-			if t.Mid() != incoming.mid {
-				continue
+			if !remoteIsPlanB {
+				// for unified plan start only the receiver marked with the matching mid
+				if t.Mid() != incomingTrackID {
+					continue
+				}
 			}
 
-			if (incomingTracks[ssrc].kind != t.kind) ||
+			if (incoming.kind != t.kind) ||
 				(t.Direction() != RTPTransceiverDirectionRecvonly && t.Direction() != RTPTransceiverDirectionSendrecv) ||
 				(t.Receiver()) == nil ||
 				(t.Receiver().haveReceived()) {
 				continue
 			}
 
-			delete(incomingTracks, ssrc)
+			delete(incomingTracks, incomingTrackID)
 			localTransceivers = append(localTransceivers[:i], localTransceivers[i+1:]...)
 			pc.startReceiver(incoming, t.Receiver())
 			break
 		}
 	}
 
 	if remoteIsPlanB {
-		for ssrc, incoming := range incomingTracks {
-			t, err := pc.AddTransceiverFromKind(incoming.kind, RtpTransceiverInit{
-				Direction: RTPTransceiverDirectionSendrecv,
-			})
-			if err != nil {
-				pc.log.Warnf("Could not add transceiver for remote SSRC %d: %s", ssrc, err)
-				continue
+		for _, incoming := range incomingTracks {
+			for ssrc := range incoming.ssrcStreams {
+				t, err := pc.AddTransceiverFromKind(incoming.kind, RtpTransceiverInit{
+					Direction: RTPTransceiverDirectionSendrecv,
```

Comment:
What if it's recv only in the remote desc (pardon my ignorance towards plan b vs unified)?

Reply:
I just kept the same logic as before adapting to the new trackDetails format. So it'll work as before. IMHO I'll just remote planb...